provisioning_client/adapters/hsm_client_riot.c

```diff
@@ -551,6 +548,15 @@ void hsm_client_riot_destroy(HSM_CLIENT_HANDLE handle)
         /* Codes_SRS_HSM_CLIENT_RIOT_07_008: [ hsm_client_riot_destroy shall free the HSM_CLIENT_HANDLE instance. ] */
         free(x509_client->certificate_common_name);
         /* Codes_SRS_HSM_CLIENT_RIOT_07_009: [ hsm_client_riot_destroy shall free all resources allocated in this module. ] */
+        mbedtls_ecp_point_free(&x509_client->ca_root_pub);
+        mbedtls_mpi_free(&x509_client->ca_root_priv);
+
+        mbedtls_ecp_point_free(&x509_client->device_id_pub);
+        mbedtls_mpi_free(&x509_client->device_id_priv);
+
+        mbedtls_ecp_point_free(&x509_client->alias_key_pub);
+        mbedtls_mpi_free(&x509_client->alias_key_priv);
```

Comment:
Previous comments apply here as well

Reply:
These are the correct calls according to the MSR team